zetaclient/chains/zrepo/zrepo.go

```diff
@@ -1,34 +1,85 @@
 // Package zrepo provides an abstraction layer for interactions with the zetacore client.
+//
+// The functions inside this module return very descriptive errors.
+// There is no need to wrap these errors with "failed to..." messages.
 package zrepo
 
 import (
 	"context"
+	"fmt"
+
+	cometbft "github.com/cometbft/cometbft/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+	grpccodes "google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
 
 	"github.com/zeta-chain/node/pkg/chains"
 	"github.com/zeta-chain/node/x/crosschain/types"
+	crosschain "github.com/zeta-chain/node/x/crosschain/types"
+	fungible "github.com/zeta-chain/node/x/fungible/types"
+	observer "github.com/zeta-chain/node/x/observer/types"
+	"github.com/zeta-chain/node/zetaclient/logs"
+	"github.com/zeta-chain/node/zetaclient/mode"
+	"github.com/zeta-chain/node/zetaclient/zetacore"
 )
 
 // ZetaRepo implements the Repository pattern by wrapping a zetacore client.
 // Each chain module must instantiate its own ZetaRepo.
 type ZetaRepo struct {
-	client ZetacoreClient
+	client ZetacoreClientRepo
 
 	connectedChain chains.Chain
+
+	clientMode mode.ClientMode
 }
 
 // New constructs a new ZetaRepo object.
-func New(client ZetacoreClient, connectedChain chains.Chain) *ZetaRepo {
-	return &ZetaRepo{client, connectedChain}
+func New(client ZetacoreClient, connectedChain chains.Chain, clientMode mode.ClientMode) *ZetaRepo {
+	return &ZetaRepo{client, connectedChain, clientMode}
+}
+
+// ------------------------------------------------------------------------------------------------
+// Getters
+// ------------------------------------------------------------------------------------------------
+
+func (repo *ZetaRepo) ZetaChain() chains.Chain {
+	return repo.client.Chain()
 }
 
 func (repo *ZetaRepo) GetCCTX(ctx context.Context, nonce uint64) (*types.CrossChainTx, error) {
 	cctx, err := repo.client.GetCctxByNonce(ctx, repo.connectedChain.ChainId, nonce)
 	if err != nil {
-		return nil, newClientError(ErrClientGetCCTXByNonce, err)
+		outterErr := fmt.Errorf("%w %d", ErrClientGetCCTX, nonce)
+		return nil, newClientError(outterErr, err)
```

Comment:
Also wondering if this would make the error repititave, 
The inner function returns `failed to get cctx by nonce` already

Reply:
This error would still have both `failed to get cctx by nonce` and `failed to get CCTX for nonce` in the error statement ?
Wondering wether we should remove one ?